### PR TITLE
Random pairs

### PR DESCRIPTION
**Pull request by fweik**
Added test to check pair finding with a random position set.

### CODE REVIEW

**Comment by fweik** [comment]
@KaiSzuttor

File: src/core/debug.cpp [hunk 1]
```diff
@@ -222,3 +222,17 @@ void check_particles() {
 
   CELL_TRACE(fprintf(stderr, "%d: leaving check_particles\n", this_node));
 }
+
+void check_particle_sorting() {
+  for (int c = 0; c < local_cells.n; c++) {
```
Comment (KaiSzuttor) [inline src/core/debug.cpp]: please add some comments what is being checked here

File: src/core/domain_decomposition.cpp [hunk 2]
```diff
@@ -635,13 +635,12 @@ int dd_append_particles(ParticleList *pl, int fold_dir) {
...
-      cpos[dir] =
-          (int)((pl->part[p].r.p[dir] - my_left[dir]) * dd.inv_cell_size[dir]) +
-          1;
+      auto lpos = pl->part[p].r.p[dir] - my_left[dir];
+      cpos[dir] = static_cast<int>(std::floor(lpos * dd.inv_cell_size[dir])) + 1;
 
       if (cpos[dir] < 1) {
         cpos[dir] = 1;
-        if (PERIODIC(dir)) {
+        if (PERIODIC(dir) || !boundary[2 * dir]) {
```
Comment (KaiSzuttor) [inline src/core/domain_decomposition.cpp]: what is this check for?

File: src/core/domain_decomposition.cpp [hunk 3]
```diff
@@ -665,10 +664,14 @@ int dd_append_particles(ParticleList *pl, int fold_dir) {
     }
     c = get_linear_index(cpos[0], cpos[1], cpos[2], dd.ghost_cell_grid);
     CELL_TRACE(fprintf(stderr,
-                       "%d: dd_append_particles: Appen Part id=%d to cell %d\n",
-                       this_node, pl->part[p].p.identity, c));
+                       "%d: dd_append_particles: Appen Part id=%d to cell %d cpos %d %d %d\n",
```
Comment (KaiSzuttor) [inline src/core/domain_decomposition.cpp]: typo

File: testsuite/random_pairs.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
+from __future__ import print_function
```
Comment (KaiSzuttor) [inline testsuite/random_pairs.py:1]: autopep8 -i on that file

File: testsuite/random_pairs.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import unittest as ut
+import espressomd
+import numpy as np
+import itertools
+import collections
+
+class RandomPairTest(ut.TestCase):
```
Comment (KaiSzuttor) [inline testsuite/random_pairs.py]: docstring for the test class

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]

**Comment by fweik** [comment]
@KaiSzuttor Fixed the issues.

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**